utils.py

```diff
@@ -0,0 +1,380 @@
+__author__ = 'Michail Xyntarakis'
```

Comment:
@e-lo I'll review them.  At least some are necessary; I added them when I started running into missing dependencies, and they were all from this set.